server/services/store/sqlstore/migrate.go

```diff
@@ -213,6 +225,194 @@ func (s *SQLStore) Migrate() error {
 	return engine.ApplyAll()
 }
 
+func (s *SQLStore) migrateSchemaVersion(migrations []*models.Migration) error {
+	migrationNeeded, err := s.isSchemaMigrationNeeded()
+	if err != nil {
+		return err
+	}
+
+	if !migrationNeeded {
+		return nil
+	}
+
+	s.logger.Info("Migrating schema migration to new format")
+
+	legacySchemaVersion, err := s.getLegacySchemaVersion()
+	if err != nil {
+		return err
+	}
+
+	if err := s.createTempSchemaTable(); err != nil {
+		return err
+	}
+
+	if err := s.populateTempSchemaTable(migrations, legacySchemaVersion); err != nil {
+		return err
+	}
+
+	if err := s.useNewSchemaTable(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) isSchemaMigrationNeeded() (bool, error) {
+	// Check if `dirty` column exists on schema version table.
+	// This column exists only for the old schema version table.
+
+	// SQLite needs a bit of a special handling
+	if s.dbType == model.SqliteDBType {
+		return s.isSchemaMigrationNeededSQLite()
+	}
+
+	query := s.getQueryBuilder(s.db).
+		Select("count(*)").
+		From("information_schema.COLUMNS").
+		Where(sq.Eq{
+			"TABLE_NAME":  s.tablePrefix + "schema_migrations",
+			"COLUMN_NAME": "dirty",
+		})
+
+	row := query.QueryRow()
+
+	var count int
+	if err := row.Scan(&count); err != nil {
+		s.logger.Error("failed to check for columns of schema_migrations table", mlog.Err(err))
+		return false, err
+	}
+
+	return count == 1, nil
+}
+
+func (s *SQLStore) isSchemaMigrationNeededSQLite() (bool, error) {
```

Comment:
Out of curiosity since I didn't worked too much with SQLite; What if we try to read from the table and try get the `dirty` value even if the column does not exists? Wouldn't it fail? Maybe we can understand that way..

Reply:
What difference would it make? Any issue with the current implementation?